### ISSUE TITLE
Extracting parameters from request 

### ISSUE THREAD

**Issue by arnaudbriche** [ngmoco/falcore#51]
Is it possible to extract parameters with the router to pass to a pipeline stage ?

**Comment by smw1218** [comment]
Yes.  Since SelectPipeline(req *Request) (pipe RequestFilter) has access to the request, you can do any sort of parsing and saving of parameters in the Context on the request.  You can see an example here:

https://github.com/ngmoco/pat/blob/master/falcore_mux.go#L23

**Comment by dgrijalva** [comment]
If you want to do param parsing, but not routing, you can simply use a RequestFilter.  Requests have a Context property you can use to store data you want to be accessible to later staging.  For example, parsed query information or authentication state data.

**Comment by arnaudbriche** [comment]
Thx, not sure I totallyget it.
Is it possible to use pat in conjonction with falcore to get both pipelines functionality and restful routing ?
Not sure it's clear enough; i'm currently using https://github.com/drone/routes to do the routing for my app; it looks as follows

```
router := routes.New()
router.Filter(Authenticate)
router.Get("/", RootHandler) 

campaignsController := new(campaigns.CampaignsController)

router.Get ("/campaigns", func(resp http.ResponseWriter, req *http.Request) { campaignsController.List(resp, req)   })
router.Post("/campaigns", func(resp http.ResponseWriter, req *http.Request) { campaignsController.Create(resp, req) })
router.Get ("/campaigns/:id", func(resp http.ResponseWriter, req *http.Request) { campaignsController.Read(resp, req)   })
router.Put ("/campaigns/:id", func(resp http.ResponseWriter, req *http.Request) { campaignsController.Update(resp, req) })
router.Del ("/campaigns/:id", func(resp http.ResponseWriter, req *http.Request) { campaignsController.Delete(resp, req) })

http.Handle("/", router)
```

But I'm missing the "after filter" functionality. 
Is it possible to do the same with falcore while getting all the cool stuff it adds up ?

**Comment by smw1218** [comment]
The ngmoco/pat fork will do restful routes in the same way as your example from drones/routes.

The one thing that is a major difference is the signature for a falcore RequestFilter is quite different from the standard library.  We wrap the http.Request in a falcore.Request and also return an http.Response rather than using the http.ResponseWriter.  It shouldn't be difficult to convert your controllers to use the falcore signature instead.
